src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -1969,4 +1976,90 @@ public Response getBannerMessages(@PathParam("id") Long id) throws WrappedRespon
 
     }
     
+    @GET
+    @Path("/licenses")
+    public Response getLicenses() {
+        JsonArrayBuilder arrayBuilder = Json.createArrayBuilder();
+        for(License license : licenseService.listAll()) {
+            arrayBuilder.add(JsonPrinter.json(license));
+        }
+        return ok(arrayBuilder);
+    }
+
+    @GET
+    @Path("/licenses/{id}")
+    public Response getLicenseById(@PathParam("id") long id) {
+        try {
+            License license = licenseService.getById(id);
+            return ok(json(license));
+        } catch (FetchException e) {
+            return error(Response.Status.NOT_FOUND, e.getMessage());
+        }
+    }
+
+    @POST
+    @Path("/licenses")
+    public Response addLicense(License license) {
+        try {
+            licenseService.save(license);
+            return created("/api/admin/licenses", Json.createObjectBuilder().add("message", "License created"));
+        } catch (RequestBodyException e) {
+            return error(Response.Status.BAD_REQUEST, e.getMessage());
+        } catch(ConflictException e) {
+            return error(Response.Status.CONFLICT, e.getMessage());
+        }
+    }
+
+	@GET
+	@Path("/licenses/default")
+	public Response getDefault(){
+		return ok("Default license ID is " + licenseService.getDefault().getId());
+	}
+
+    @PUT
+    @Path("/licenses/default")
+    public Response setDefault(long id) {
+        try {
+            licenseService.setDefault(id);
+            return ok("Default license ID set to " + id);
+        }
+        catch (UpdateException | FetchException e) {
+            return error(Status.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+        catch (IllegalArgumentException e) {
+            return badRequest(e.getMessage());
+        }
+    }
+
+    @PUT
+    @Path("/licenses/{id}")
+    public Response putLicenseById(@PathParam("id") long id, License license) {
```

Comment:
same question as in the licenseService.setById method - does reusing the id make sense? It gives you a way to update the license for all existing dataset versions that use it, but that seems like a dangerous thing - at a minimum I'd suggest requiring superadmin privs if this functionality you need.

Reply:
Makes sense to remove this. The thinking was that you would want to be able to correct the license description after it was referenced, but I agree now that avoiding accidental or malicious replacement of licenses is more important. I'm fine with removing this method and endpoint.